internal/runners/packages/info.go

```diff
@@ -113,67 +136,145 @@ func specificIngredientVersion(ingredientID *strfmt.UUID, version string) (*inve
-	Versions        []string `locale:"," json:"versions"`
+	name                 string
+	version              string
+	plainVersions        []string
+	PkgVersionVulnsTotal int      `opts:"omitEmpty" locale:"package_vulnerabilities,[HEADING]Vulnerabilities[/RESET]"`
+	PkgVersionVulns      []string `opts:"omitEmpty" locale:"package_cves,[HEADING]CVEs[/RESET]"`
+	*PkgDetailsTable     `locale:"," opts:"verticalTable,omitEmpty"`
+	Versions             []string `locale:"," json:"versions"`
 }
 
-func newInfoResult(ingredient *inventory_models.Ingredient, ingredientVersion *inventory_models.IngredientVersion, authors model.Authors, versions []*inventory_models.SearchIngredientsResponseVersion) *infoResult {
+func newInfoResult(so structuredOutput) *infoResult {
 	res := infoResult{
-		name:    locale.T("unknown_value"),
-		version: locale.T("unknown_value"),
-		PkgDetailsTable: PkgDetailsTable{
-			Website:   locale.T("unknown_value"),
-			copyright: locale.T("unknown_value"),
-			license:   locale.T("unknown_value"),
-		},
+		PkgDetailsTable: &PkgDetailsTable{},
 	}
 
-	if ingredient.Name != nil {
-		res.name = *ingredient.Name
+	if so.Ingredient.Name != nil {
+		res.name = *so.Ingredient.Name
 	}
 
-	if ingredient.Description != nil {
-		res.Description = *ingredient.Description
+	if so.IngredientVersion.Version != nil {
+		res.version = *so.IngredientVersion.Version
 	}
 
-	website := ingredient.Website.String()
-	if website != "" {
-		res.PkgDetailsTable.Website = website
+	if so.Ingredient.Description != nil {
+		res.PkgDetailsTable.Description = *so.Ingredient.Description
 	}
 
-	if ingredientVersion.Version != nil {
-		res.version = *ingredientVersion.Version
+	if so.Ingredient.Website != "" {
+		res.PkgDetailsTable.Website = so.Ingredient.Website.String()
 	}
 
-	if ingredientVersion.CopyrightText != nil {
-		res.PkgDetailsTable.copyright = *ingredientVersion.CopyrightText
+	if so.IngredientVersion.LicenseExpression != nil {
+		res.PkgDetailsTable.License = *so.IngredientVersion.LicenseExpression
 	}
 
-	if ingredientVersion.LicenseExpression != nil {
-		res.PkgDetailsTable.license = *ingredientVersion.LicenseExpression
+	for _, version := range so.Versions {
+		res.plainVersions = append(res.plainVersions, version.Version)
 	}
 
-	for _, version := range versions {
-		res.Versions = append(res.Versions, version.Version)
+	if len(so.Authors) == 1 {
+		if so.Authors[0].Name != nil {
+			res.Author = fmt.Sprintf("[ACTIONABLE]%s[/RESET]", *so.Authors[0].Name)
+		}
+	} else if len(so.Authors) > 1 {
+		var authorsOutput []string
+		for _, author := range so.Authors {
+			if author.Name != nil {
+				authorsOutput = append(authorsOutput, *author.Name)
+			}
+		}
+		res.Authors = fmt.Sprintf("[ACTIONABLE]%s[/RESET]", strings.Join(authorsOutput, ", "))
 	}
 
-	for _, author := range authors {
-		if author.Name != nil {
-			res.Authors = append(res.Authors, *author.Name)
+	if len(so.Vulnerabilities) > 0 {
```

Comment:
I can't comment the whole block cause it's so big it requires scrolling.. but this whole block feels like it is doing a lot of interpretation that we could probably do in the model. So that this code only deals with representing the info, not interpreting.

That'll come in handy also with the other stories that show this type of information.

Another idea might be to turn this into a runbit. Since again showing the quick cve summary is something we'll want to do in other commands.

Reply:
Decided to go with processing this information in the model package. This code block is still fairly large but it's using the already organized information to build up the output.